src/decimal.rs

```diff
@@ -1514,6 +1514,122 @@ impl Decimal {
             flags: flags(self.is_sign_negative(), quotient_scale),
         })
     }
+
+    /// The estimated exponential function, e<sup>x</sup>, rounded to 8 decimal places. Stops
+    /// calculating when it is within tolerance is roughly 0.000002 in order to prevent
+    /// multiplication overflow.
+    fn exp(&self) -> Decimal {
+        if self == &Decimal::zero() {
+            return Decimal::one();
+        }
+
+        let tolerance = Decimal::new(2, 7);
+        let mut term = self.clone();
+        let mut result = self.clone() + Decimal::one();
+        let mut prev_result = Decimal::zero();
+        let mut factorial = Decimal::one();
+        let mut n = Decimal::new(2, 0);
+
+        // Needs rounding because multiplication overflows otherwise.
+        while (result - prev_result).abs() > tolerance && n < Decimal::new(24, 0) {
+            prev_result = result;
+            term = self * term.round_dp(8);
+            factorial *= n;
+            result += (term / factorial).round_dp(8);
+            n += Decimal::one();
+        }
+
+        result
+    }
+
+    /// Raise self to the given unsigned integer exponent: x<sup>y</sup>
+    fn powi(&self, exp: u64) -> Decimal {
+        if exp == 0 {
+            Decimal::one()
+        } else if exp == 1 {
+            self.clone()
+        } else {
+            let mut result = self.clone();
+            for _ in 1..exp {
+                result *= self;
+            }
+            result
+        }
+    }
+
+    /// The square root of a Decimal. Uses a standard Babylonian method.
+    fn sqrt(&self) -> Option<Decimal> {
+        if self.is_sign_negative() {
+            return None;
+        }
+
+        if self == &Decimal::zero() {
+            return Some(Decimal::zero());
+        }
+
+        // Start with an arbitrary number as the first guess
+        let mut result = self / Decimal::new(2, 0);
+        let mut last = result + Decimal::one();
+
+        // Keep going while the difference is larger than the tolerance
+        while last != result {
+            last = result;
+            result = (result + self / result) / Decimal::new(2, 0);
+        }
+
+        return Some(result);
+    }
+
+    /// The natural logarithm for a Decimal. Uses a [fast estimation algorithm](https://en.wikipedia.org/wiki/Natural_logarithm#High_precision)
+    /// This is more accurate on larger numbers and less on numbers less than 1.
+    fn ln(&self) -> Decimal {
+        if self.is_sign_positive() {
+            if self == &Decimal::one() {
+                Decimal::zero()
+            } else {
+                let pi = Decimal::from_str("3.141592653589793238462643383279502884197169399375105820974").unwrap();
```

Comment:
This could be a useful public constant if we're deciding to define it. We could then avoid the string unwrap/parsing.

Reply:
I aded this as a constant, along with `TWO`. It doesn't speed up too much, but it does help.